### PR TITLE
Feature: Provided support for django 4.X

### PR DESCRIPTION
**Pull request by onegreyonewhite**
##### Changelog:
* Feature(CI): Add gh-actions tests.
* Feature(tests): Add coverage in tox tests.
* Fix: Import error with Django 4.1.
* Fix: Drop old django versions and fix tox tests.
* Fix: Tests on django 4.0 and 4.1.
* Chore(dev): Add default test configs for pycharm.

### CODE REVIEW

**Comment by onegreyonewhite** [comment]
@littlepea may be should drop old versions of python (less 3.6) and django (less 3.2). I think it would be better to support latest supported django LTS and python from the django requirements.

I can create PR with updated specs and tox config.

**Comment by littlepea** [comment]
@onegreyonewhite yes, please! 🙏

**Comment by onegreyonewhite** [comment]
> @onegreyonewhite yes, please! pray

@littlepea but it stil WIP. django-nose deprecated and some tests fails (I don't know why). You can merge it, if you know how to fix it. And my be should migrate to github actions.

**Comment by littlepea** [comment]
I don't know how to fix it 🙂 
I'd love to migrate to GH actions, but don't have time to do anything on this repository these days...

**Comment by onegreyonewhite** [comment]
> I don't know how to fix it 🙂 
> I'd love to migrate to GH actions, but don't have time to do anything on this repository these days...

OK. I'm going to create fix tomorrow (if it possible🙃)

**Comment by onegreyonewhite** [comment]
@littlepea done. I have made tests without a detailed listing of python versions, because there is no point in testing directly on each.

File: .run/Debug tests.run.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<component name="ProjectRunConfigurationManager">
```
Comment (littlepea) [inline .run/Debug tests.run.xml:1]: I'm not sure there's `.run/*` files should be version controlled?

Comment (onegreyonewhite) [reply .run/Debug tests.run.xml:1]: @litlepeathis will greatly simplify the life of those who will develop in PyCharm.  If this is a problem, I can remove it.

Comment (littlepea) [reply .run/Debug tests.run.xml:1]: No, it's fine

**Review by littlepea — APPROVED** [review]

**Comment by littlepea** [comment]
Looks like the build failed though: https://github.com/littlepea/django-docs/actions/runs/3541796156/jobs/5946508285

**Comment by onegreyonewhite** [comment]
> Looks like the build failed though: https://github.com/littlepea/django-docs/actions/runs/3541796156/jobs/5946508285

@littlepea I don't know why. This works fine [in my repo](https://github.com/onegreyonewhite/django-docs/actions/runs/3541830030).

**Comment by onegreyonewhite** [comment]
@littlepea can I help with anything, which can move release date early? ;)

**Comment by littlepea** [comment]
Published a new version: https://pypi.org/project/django-docs/0.3.3/

**Comment by littlepea** [comment]
@onegreyonewhite Btw, are you interested to be one of the maintainers of this package?

**Comment by onegreyonewhite** [comment]
> @onegreyonewhite Btw, are you interested to be one of the maintainers of this package?

@littlepea need to think. We need investing in time to rewrite the tests at least for django unittest here.  It would be better to automate the rollout of releases.  Then you can work on the project in our free time.

**Pull request merged**